### PR TITLE
Update outdated Issue Templates to reflect our current process

### PR DESCRIPTION
**Pull request by chachasikes**
In our early stages of Design System work, I created some issue templates as a way to describe our process.
Now that we have iterated on our process more, and about to do a presentation about this project and show our GitHub and how we are working in the open, I've updated our templates.

- Cleaned up the Proposal template based on conversations
- Added some basic issue templates
- Adjusted the order of the templates (as they appear when you visit the Github Add new issue page. 
- Included a support template because our form error message needs a fallback, and Github can be our fallback (though it requires a login, last we checked)
- Adjusted the language on the templates
- Adjusted the headers

At the very least, these template give some visibility into the kinds of reviews we do on our components at different stages - and can be a reference for a PM, or an independent team who wants to understand how to contribute.

@TODO 
* Document our baselines for accessibility, translations, and performance reviews.
* Validate the UX review with Adam
* Improve our WIKI with some of the information in the issue templates.
* Update, clean up project labels
* Decide if we want any assignees - like perhaps a governance team in GitHub with assignees in the team?
* Test the templates
* Technical review of our checklists
* Link to any baseline docs we create (in the wiki or on the website)
* Link to our contributing page

### CODE REVIEW

**Comment by chachasikes** [comment]
Doing a first round of review with ODI engineers and Art/Koji (for DWSN presentation)

Then will share with some others

File: .github/ISSUE_TEMPLATE/design-system-04-registration.md [hunk 1]
```diff
@@ -0,0 +1,56 @@
+---
+name: Component registration
```
Comment (zakiya) [inline .github/ISSUE_TEMPLATE/design-system-04-registration.md:2]: Maybe "New component registration" ? I had to scroll through to figure out exactly what this is.

**Review by zakiya — APPROVED** [review]
Awesome work.

File: .github/ISSUE_TEMPLATE/design-system-04-registration.md (hunk 1 above)
Comment (chachasikes) [reply .github/ISSUE_TEMPLATE/design-system-04-registration.md:2]: Great suggestion - thanks! I will add this change in next rev

**Pull request merged**